e2eAIOK/deltatuner/deltatuner/scores/transformer_proxy.py

```diff
@@ -0,0 +1,92 @@
+import os, sys, time, math
+import torch
+from torch import nn
+import numpy as np
+import gc
+import torch
+from torchsummary import summary
+from torch.nn.modules import linear
+from transformers import pytorch_utils
+
+from .utils import ATTN_LAYER_STRUCTURE_NAME, LINEAR_LAYER_STRUCTURE_NAME, LINEAR_MODULE_TYPE, LINEAR_LORA_MODULE_TYPE
+
+# TODO: currently specific support for one pure lora layer
+def network_weight_gaussian_init(model):
+    with torch.no_grad():
+        for k, m in model.named_modules():
+            if any(isinstance(m, module_type) for module_type in LINEAR_LORA_MODULE_TYPE):
+                if hasattr(m, "lora_A") and "default" in m.lora_A.keys():
+                    nn.init.kaiming_uniform_(m.lora_A["default"].weight, a=math.sqrt(5), mode='fan_in')
+                    if hasattr(m.lora_A["default"], "bias") and m.lora_A["default"].bias is not None:
+                        nn.init.zeros_(m.lora_A["default"].bias)
+                if hasattr(m, "lora_B") and "default" in m.lora_B.keys():
+                    nn.init.kaiming_uniform_(m.lora_B["default"].weight, a=math.sqrt(5), mode='fan_out')
+                    if hasattr(m.lora_B["default"], "bias") and m.lora_B["default"].bias is not None:
+                        nn.init.zeros_(m.lora_B["default"].bias)
+    return model
+
+# TODO: separate domain specific ops
+def get_linear_layer_metric_array(net, metric):
```

Comment:
looks those functions are the same as transfomer_proxy_ssf.py, can it be moved to basic_util?

Reply:
Some of them could be moved to basic_util, it need some effort, after checking with @XinyaoWa and @tianyil1 , we decide to refine it in next PR.